Source/LinqToDB/Data/DataConnection.cs

```diff
@@ -1639,12 +1640,12 @@ public virtual void RollbackTransaction()
 			{
 				var actionResult = action(dataConnection, context);
 
-				if (TraceSwitchConnection.TraceInfo)
+				if (dataConnection.TraceSwitchConnection.TraceInfo)
 				{
-					OnTraceConnection(new TraceInfo(this, TraceInfoStep.AfterExecute, traceOperation, false)
+					dataConnection.OnTraceConnection(new TraceInfo(dataConnection, TraceInfoStep.AfterExecute, traceOperation, false)
 					{
 						TraceLevel    = TraceLevel.Info,
-						CommandText   = commandText,
+						CommandText   = dataConnection.TraceSwitchConnection.TraceInfo ? sql : commandText?.Invoke(context),
```

Comment:
* We already know `dataConnection.TraceSwitchConnection.TraceInfo == true` due to the `if()`
* We know that when it is `true`, the value is already set in `sql`.

```suggestion
						CommandText   = sql,
```